l1-contracts/src/core/Rollup.sol

```diff
@@ -77,6 +89,9 @@ contract Rollup is EIP712("Aztec Rollup", "1"), Leonidas, IRollup, ITestRollup {
   //
   //        More direct approach would be storing keccak256(header) as well
   mapping(uint256 blockNumber => BlockLog log) public blocks;
+  // The below public inputs are filled when proposing a block, then used to verify an epoch proof.
+  // TODO(#8955): When implementing batched kzg proofs, store one instance per epoch rather than block
+  mapping(uint256 blockNumber => BlobPublicInputs) public blobPublicInputs;
```

Comment:
This is going to be insanely expensive in storage, 4 storage slots per blob. You can do it for now and leave the gas optimisation for tmnt. #9826